tsdb/engine/tsm1/encoding.go

```diff
@@ -111,26 +111,45 @@ func (a Values) Merge(b Values) Values {
 		return a
 	}
 
-	var i, j int
-	for ; i < len(a) && j < len(b); i++ {
-		av, bv := a[i].UnixNano(), b[j].UnixNano()
+	for i := 0; i < len(a) && len(b) > 0; i++ {
+		av, bv := a[i].UnixNano(), b[0].UnixNano()
+		// Value in a is greater than B, we need to merge
 		if av > bv {
-			a[i], b[j] = b[j], a[i]
+			// Save value in a
+			temp := a[i]
+
+			// Overwrite a with b
+			a[i] = b[0]
+
+			// Slide all values of b down 1
+			copy(b, b[1:])
+			b = b[:len(b)-1]
```

Comment:
Doh of course. There will always be the extra capacity needed to avoid the allocation...